### ISSUE TITLE
Add ability to customize the options passed to `createHistory`

### ISSUE THREAD

**Issue by PAkerstrand** [rocjs/roc-package-web-app-react#31]
TL;DR: We would like a way to customize the options passed to `createHistory`.
## Background

In our hybrid apps, we run into issues when our app uses single page navigation, since the apps don't get triggers that navigation events have occurred. This causes them to behave erratically and is very annoying for the users. In these environments, we'd like to force a reload, similar to how IE9 and 10 is handled. `createBrowserHistory` allows a `forceRefresh`-option that can be used to do exactly this.
## Suggestion
- Add ability to configure the options passed to `createHistory`. These will be merged together with the `basename` option that is currently passed.
## Considerations
- `createBrowserHistory` and `createMemoryHistory` share some configuration options, such as `keyLength`, that will probably need to be the same in order to avoid hydration-issues
- Should we allow `basename` to be overridden here, or should the option in `createClient` take precedence?

**Comment by dlmr** [comment]
Make sense that it should be possible to modify this, so I'm for this change. 👍 

`basename` is something that I do not believe that we need to expose, however I think it's is a good idea that it can be changed through `__raw`.
